tests/test_model.py

```diff
@@ -522,3 +522,417 @@ def test_model_diskonlyfilemodel_field_is_constructed_correctly(
+        save_location: Path = tmp_path / "dimr_config.xml"
+        dimr.save(filepath=save_location)
+
+        process = "<process>"
+
+        with open(save_location, "r") as file:
+            assert (
+                process not in file
+            ), f"File {save_location} does contain the line: {process}"
+
+
+class TestFmComponentProcess:
+    @pytest.mark.parametrize(
+        "input_process",
+        [
+            pytest.param(None),
+            pytest.param(1),
+            pytest.param(2),
+            pytest.param(3),
+            pytest.param(4),
+            pytest.param(5),
+        ],
+    )
+    def test_fmcomponent_process_after_init(self, input_process: int):
+        component = FMComponent(
+            name="test",
+            workingDir=".",
+            inputfile="test.mdu",
+            process=input_process,
+            mpiCommunicator="DFM_COMM_DFMWORLD",
+        )
+        assert component.process == input_process
+
+    @pytest.mark.parametrize(
+        "input_process",
+        [
+            pytest.param(None),
+            pytest.param(1),
+            pytest.param(2),
+            pytest.param(3),
+            pytest.param(4),
+            pytest.param(5),
+        ],
+    )
+    def test_fmcomponent_process_after_setting(self, input_process: int):
+        component = FMComponent(
+            name="test",
+            workingDir=".",
+            inputfile="test.mdu",
+            mpiCommunicator="DFM_COMM_DFMWORLD",
+        )
+        component.process = input_process
+        assert component.process == input_process
+
+    def test_fmcomponent_without_process_after_init(self):
+        expected_process_format = None
+        component = FMComponent(
+            name="test",
+            workingDir=".",
+            inputfile="test.mdu",
+            mpiCommunicator="DFM_COMM_DFMWORLD",
+        )
+        assert component.process == expected_process_format
+
+    @pytest.mark.parametrize(
+        "input_process",
+        [
+            pytest.param("lalala"),
+            pytest.param("123"),
+            pytest.param(3.5),
+        ],
+    )
+    def test_fmcomponent_process_after_init_with_incorrect_input_throws_valueerror(
+        self,
+        input_process,
+    ):
+        with pytest.raises(ValueError) as error:
+            FMComponent(
+                name="test",
+                workingDir=".",
+                inputfile="test.mdu",
+                process=input_process,
+                mpiCommunicator="DFM_COMM_DFMWORLD",
+            )
+
+        expected_message = (
+            f"In component 'test', the keyword process '{input_process}', is incorrect."
+        )
+        assert expected_message in str(error.value)
+
+    @pytest.mark.parametrize(
+        "input_process",
+        [
+            pytest.param(0),
+            pytest.param(-1),
+        ],
+    )
+    def test_fmcomponent_process_after_init_with_input_process_zero_throws_valueerror(
+        self,
```

Comment:
These kind of unit tests should be moved to `\tests\dimr\test_dimr.py` instead of putting them at this high level integration tests file.

Reply:
Lets have a call where we want to place which test class of the two new test classes.

I had not found any other Component or FMcomponent test.
Thus I was unsure if I should have created a new test file or keep it on the same level as where DIMR() is tested

The `\tests\dimr\test_dimr.py` does not have any test that inits `DIMR`